docs/0005-hashed-n-tuple-id-layout.md

```diff
@@ -0,0 +1,235 @@
+# OCFL Community Extension 0005: Hashed Truncated N-tuple Trees with Encapsulating Directory for OCFL Storage Hierarchies
+
+  * **Extension ID:** 0005-hashed-n-tuple-id-layout
+  * **Authors:** Ben Cail
+  * **Minimum OCFL Version:** 1.0
+  * **Obsoletes:** n/a
+  * **Obsoleted by:** n/a
+
+## Overview
+
+This storage root extension describes how to safely map OCFL object identifiers containing any characters to OCFL object root directories.
+
+Using this extension, OCFL object identifiers are hashed and encoded as hex strings (all letters lower-case). These digests are then divided into _N_ n-tuple segments, which are used to create nested paths under the OCFL storage root. Finally, the OCFL object identifier is URL-encoded to create a directory name for the OCFL object root.
```

Comment:
In unix systems, most characters can be used in filenames without causing problems. An obvious exception is `/`. Do you think it's worthwhile to have a parameter for disabling the percent-encoding for people who know their ids won't cause problems and don't want the added bloat?

Reply:
I think I'd rather leave the encoding always enabled, unless there's someone with a strong request to use this layout in their repo, with identifier characters that would be encoded, and not wanting them encoded in the directory name.